server/gfmd/user.c

```diff
@@ -298,14 +298,22 @@ user_enter_gsi_dn(const char *gsi_dn, struct user *u)
 
 static struct user *
 user_lookup_auth_id_including_invalid(
-	enum auth_user_id_type auth_user_id_type,
+	const char *auth_user_id_type_str,
 	const char *auth_user_id)
 {
 	struct gfarm_hash_entry *entry;
+	enum auth_user_id_type auth_user_id_type;
 
 	if (user_is_null_str(auth_user_id))
 		return (NULL);
 
+	if (gfarm_auth_user_id_type_from_name(
```

Comment:
calling gfarm_auth_user_id_type_from_name() here is a waste of CPU time.
instead of changing the first parameter of user_lookup_auth_id_including_invalid() from enum auth_user_id_type to const char *, please keep the parameter type and let the caller directly pass enum auth_user_id_type, by exporting enum auth_user_id_type in user.h.